### PR TITLE
fix: rb command regression deleting bucket

### PR DESCRIPTION
**Pull request by harshavardhana**

### CODE REVIEW

File: cmd/rb-main.go [hunk 1]
```diff
@@ -235,7 +235,10 @@ func mainRemoveBucket(cliCtx *cli.Context) error {
 		}
 
 		listCtx, listCancel := context.WithCancel(ctx)
-		for range clnt.List(listCtx, opts) {
+		for obj := range clnt.List(listCtx, opts) {
+			if obj.Err != nil {
+				continue
```
Comment (vadmeste) [inline cmd/rb-main.go:240]: Good catch, but there is no point continuing here: if we are going to delete a bucket, we should list all of its contents and remove them. fatalIf() is enough.

**Comment by vadmeste** [comment]
The fix is needed.. but just not sure how to reproduce an issue with this.

**Comment by harshavardhana** [comment]
> The fix is needed.. but just not sure how to reproduce an issue with this.

@vadmeste FS mode

File: cmd/rb-main.go (hunk 1 above)
Comment (harshavardhana) [reply cmd/rb-main.go:240]: continue is to look for situations if there are objects @vadmeste such that we don't prematurely send DeleteBucket()

**Comment by vadmeste** [comment]
> @vadmeste FS mode

oh okay.. maybe FS mode should support versioning listing as well.

**Comment by harshavardhana** [comment]
> > @vadmeste FS mode
> 
> oh okay.. maybe FS mode should support versioning listing as well.

No it shouldn't @vadmeste

**Pull request merged**